src/hocs/WithTranslation/WithTranslation.jsx

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import { connect } from 'react-redux';
+
+import engLanguagePack from '../../intl/en.json';
+import uaLanguagePack from '../../intl/ua.json';
+import { localizationSelector } from '../../redux/movies/movies.selectors';
+import { languageShortages } from '../../constants';
+
+const WithTranslation = (WrappedComponent, name) => {
+  const mapStateToProps = (state) => ({
+    currentLocalization: localizationSelector(state)
+  });
+
+  const localizations = {
+    [languageShortages.ENGLISH]: engLanguagePack,
+    [languageShortages.UKRAINIAN]: uaLanguagePack
+  };
+
+  const ComponentWithTranslation = ({ currentLocalization, ...otherProps }) => {
+    return (
+      <WrappedComponent localizationData={localizations[currentLocalization][name]} {...otherProps} />
+    )
+  };
+
+  return connect(mapStateToProps)(ComponentWithTranslation);
+};
+
+export default WithTranslation;
```

Comment:
1. Name of hoc. I docs you can find that your hoc should start with lowercase letter. `withTranslations`.
2. If you want to pass some parameter - `const withTranslation = (options) => (WrappedComponent) => (...)`
3. You don't need name here at all. Translations pack should be just plain object with key and value. No need in nested separate objects.
4. Name of component inside. Instead `ComponentWithTranslation` it should be `WithTranslation`, the same as hoc name, but with uppercased first letter

In addition: read, please, https://reactjs.org/docs/higher-order-components.html this doc, you can find all common rules and advices here